### PR TITLE
[TSI] Query/GetSeries client library implementation

### PR DESCRIPTION
**Pull request by barustum**
Not merging to master, rather from a private branch to another, in order to reduce the PR size. Once all of GetSeries is complete, will merge to master.

This PR does not include E2E tests. E2E tests are a work in progress.

### CODE REVIEW

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Customized/Models/GetSeries.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        /// <summary>
+        /// Time Series Id that uniquely identifies the instance. It matches Time Series Id properties in an environment.
+        /// </summary>
+        public TimeSeriesId TimeSeriesId { get; }
+
+        /// <summary>
+        /// Initializes a new instance of GetSeries.
+        /// </summary>
+        /// <param name="timeSeriesId">
+        /// Time Series Id that uniquely identifies the instance.It matches Time Series Id properties in
```
Comment (bikamani) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Customized/Models/GetSeries.cs:39]: space before "It matches ..

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Customized/Models/GetSeries.Serialization.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.IoT.TimeSeriesInsights
+{
+    /// <summary>
+    /// This class definition overrides serialization and deserialization implementation in
```
Comment (bikamani) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Customized/Models/GetSeries.Serialization.cs:10]: what is it you had to change?

**Review by bikamani — APPROVED** [review]

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Customized/Models/GetSeries.Serialization.cs (hunk 2 above)
Comment (barustum) [reply sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Customized/Models/GetSeries.Serialization.cs:10]: I changed how TimeSeriesId is being serialized. It was being serialized as a list of object, now it's being serialized as a strongly typed object. That's what I try to explain in this comment block. Let me know if unclear and I can change it.

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Models/QuerySeriesRequestOptions.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    /// <summary>
+    /// Optional parameters to use when querying for Time Series Insights series.
+    /// </summary>
+    public class QuerySeriesRequestOptions : QueryRequestOptions
+    {
+        /// <summary>
+        /// Selected variables that needs to be projected in the query result. When it is null or not set, all the
+        /// variables and Time Series types in the model are returned.
+        /// </summary>
+        public string[] ProjectedVariables { get; set; }
```
Comment (drwill-ms) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Models/QuerySeriesRequestOptions.cs:17]: Should this be a list so items are easier to add/remove?

**Review by drwill-ms — APPROVED** [review]

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Models/QuerySeriesRequestOptions.cs (hunk 3 above)
Comment (barustum) [reply sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Models/QuerySeriesRequestOptions.cs:17]: Good idea. Will change.

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Customized/Models/GetSeries.Serialization.cs (hunk 2 above)
Comment (bikamani) [reply sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Customized/Models/GetSeries.Serialization.cs:10]: Yes it is clear.

**Pull request merged**